### ISSUE TITLE
[type-annotation-spacing] Throws Error on Constructor Interface Pattern

### ISSUE THREAD

**Issue by jarretmoses** [typescript-eslint/typescript-eslint#1669]
**Repro**

I am using the default settings for this rule. ie spaces before and after arrows in functions needed and space after a colon is needed. You can recreate these rules with the following setup.

```JSON
{
  "rules": {
    '@typescript-eslint/type-annotation-spacing': [
      2,
      {
        'overrides': {
          'arrow': {
            'before': true,
            'after': true,
          },
          'colon': {
            'before': false,
            'after': true,
          }
        }
      },
    ],
  }
}
```

```TS
  interface GoogleMaps {
    Polyline: new (opts?: google.maps.PolylineOptions) => google.maps.Polyline;
    LatLngBounds: new (
      sw?: google.maps.LatLng | google.maps.LatLngLiteral,
      ne?: google.maps.LatLng | google.maps.LatLngLiteral,
    ) => google.maps.LatLngBounds;
  }
```

**Expected Result**

The rule should pass as theres a space before and after the arrows on the constructor interfaces.

**Actual Result**

`error  Unexpected space before the '=>'  @typescript-eslint/type-annotation-spacing`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.22.0` |
| `@typescript-eslint/parser`        | `2.22.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.4.1` |
| `npm`                              | `6.14.1` |

**Comment by bradzacher** [comment]
Please make sure you use the search for closed issues: https://github.com/typescript-eslint/typescript-eslint/issues/1667

There is a fix released to the canary tag, otherwise it'll be released to latest next monday.
